synapse/lib/stormlib/aha.py

```diff
@@ -1,22 +1,105 @@
+            'get': self._methSvcGet,
+            'list': self._methSvcList,
+        }
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _methSvcList(self):
+        self.runt.reqAdmin()
+        proxy = await self.runt.snap.core.reqAhaProxy()
+        # DISCUSS: Do we want this to be a generator or list?
+        svcs = []
+        async for info in proxy.getAhaSvcs():
+            svcs.append(info)
+        return svcs
+
+    async def _methSvcDel(self, name, network=None):
+        self.runt.reqAdmin()
+        name = await s_stormtypes.tostr(name)
+        network = await s_stormtypes.tostr(network, noneok=True)
+        proxy = await self.runt.snap.core.reqAhaProxy()
+        return await proxy.delAhaSvc(name, network=network)
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _methSvcGet(self, name, filters=None):
+        self.runt.reqAdmin()
+        name = await s_stormtypes.tostr(name)
+        filters = await s_stormtypes.toprim(filters)
+        proxy = await self.runt.snap.core.reqAhaProxy()
+        return await proxy.getAhaSvc(name, filters=filters)
+
 @s_stormtypes.registry.registerLib
 class AhaPoolLib(s_stormtypes.Lib):
     '''
     A Storm Library for interacting with AHA service pools.
     '''
+
+    _storm_locals = (
+        {'name': 'add', 'desc': 'Add a new AHA service pool.',
+         'type': {'type': 'function', '_funcname': '_add',
+                  'args': (
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the pool to add.', },
+                  ),
+                  'returns': {'type': 'aha:pool'}}},
+        {'name': 'del', 'desc': 'Delete an existing AHA service pool.',
+         'type': {'type': 'function', '_funcname': '_del',
+                  'args': (
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the pool to delete.', },
+                  ),
+                  'returns': {'type': 'dict', 'desc': 'The AHA pool definition that was deleted.'}}},
+        {'name': 'get', 'desc': 'Get an existing AHA service pool.',
+         'type': {'type': 'function', '_funcname': '_get',
+                  'args': (
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the pool to get.', },
+                  ),
+                  'returns': {'type': ['null', 'aha:pool'], 'desc': 'The pool if it exists, or $lib.null.'}}},
+        {'name': 'list', 'desc': 'Enumerate all of the AHA service pools.',
+         'type': {'type': 'function', '_funcname': '_list',
+                  'returns': {'name': 'yields', 'type': 'aha:pool'}}},
+    )
     _storm_lib_path = ('aha', 'pool')
 
     def getObjLocals(self):
         return {
-            'add': self.add,
+            'add': self._add,
             'del': self._del,
-            'get': self.get,
-            'list': self.list,
+            'get': self._get,
+            'list': self._list,
         }
 
-    async def add(self, name):
+    async def _add(self, name):
```

Comment:
why are we renaming these?

Reply:
visual consistency.

Reply:
renamed these to be consistent across the module.